tests/lib/rules/no-undefined.js

```diff
@@ -42,6 +46,10 @@ ruleTester.run("no-undefined", rule, {
         { code: "function f(undefined) {}", errors },
         { code: "var undefined;", errors },
         { code: "try {} catch(undefined) {}", errors },
-        { code: "(function undefined(){}())", errors }
+        { code: "(function undefined(){}())", errors },
+        { code: "undefined = true", errors },
+        { code: "var undefined = true", errors },
+        { code: "({ bar: undefined })", errors },
+        { code: "({ bar: undefined } = foo)", errors }
```

Comment:
One creates a new variable, the other assigns to an existing one (possibly a global one) - they're different :-)